velox/functions/lib/window/NthValue.cpp

```diff
@@ -265,13 +304,14 @@ class NthValueFunction : public exec::WindowFunction {
 
   const exec::WindowPartition* partition_;
 
-  // These fields are set if the offset argument is a constant value.
-  std::optional<int64_t> constantOffset_;
+  // This field is set if the offset argument is a constant value.
+  std::optional<std::variant<int32_t, int64_t>> constantOffset_;
   bool isConstantOffsetNull_ = false;
 
   // This vector is used to extract values of the offset argument column
   // (if not a constant offset value).
-  FlatVectorPtr<int64_t> offsets_;
+  std::optional<std::variant<FlatVectorPtr<int32_t>, FlatVectorPtr<int64_t>>>
```

Comment:
Lets not use variant here either. We can have a BaseVector* here and extract the type from it at runtime.